app/assets/javascripts/dateselect.js

```diff
@@ -1,6 +1,28 @@
 $(document).on('ready page:load', function() {
+
+  function enableAllTheseDays(date) {
+    var sdate = $.datepicker.formatDate( 'yy-mm-dd', date)
+    if($.inArray(sdate, gon.available_nights) != -1) {
+        return [true];
+    }
+    return [false];  
+  }
+  
   $('#Check_In').datepicker();
   $('#Check_Out').datepicker();
   $('#First_Night').datepicker();
   $('#Last_Night').datepicker();
+  $('#Couch_Listing_Check_In').datepicker({
+    beforeShowDay: enableAllTheseDays,
+    minDate: 0,
+    numberOfMonths: 3,
+    showButtonPanel: true
+  });
+  $('#Couch_Listing_Check_Out').datepicker({
+    beforeShowDay: enableAllTheseDays,
+    minDate: 0,
+    numberOfMonths: 3,
+    showButtonPanel: true
+  });
```

Comment:
This may be my lack of js jquery knowledge, Don't we want number of months to be 12 (or 11 if 0 indexed) so couches can be reserved 1 year in advance?

@njgheorghita